pkg/signalmeow/client.go

```diff
@@ -48,9 +48,12 @@ type Client struct {
 
 	AuthedWS             *web.SignalWebsocket
 	UnauthedWS           *web.SignalWebsocket
-	WSCancel             context.CancelFunc
 	lastConnectionStatus SignalConnectionStatus
 
+	ctx    context.Context
```

Comment:
Is there some reason this is stored rather than just being passed to functions like before?

Reply:
Nope! Is it nicer passing thru or stored on the client? Not sure myself, can change..

Reply:
Usually passing directly is preferred unless there's some reason it's difficult to do. No risk of the context somehow being replaced halfway through if it's passed as a parameter.

Reply:
https://go.dev/blog/context-and-structs: no storing them